batch/batch/file_system.py

```diff
@@ -0,0 +1,162 @@
+from abc import ABC, abstractmethod
+from functools import wraps
+import os
+import glob
+import google.oauth2.service_account
+
+from hailtop.utils import blocking_to_async
+
+from .google_storage import GCS
+
+
+wildcards = ('*', '?', '[', ']', '{', '}')
+
+
+# need a custom escape because escaped characters are not treated properly with glob.escape
```

Comment:
Let's package the globbing functionality into a reusable module. Let's create `hail_glob.py`. The top of that file can have a few comments describing the glob syntax that you're implementing. It can expose two functions: `match(string, pattern)`  and `local_files(pattern)`.

Those functions are thin wrappers around what you've already built here:

```python3
def match(string, pattern):
    return fnmatch.fnmatchcase(string, escape(pattern))

def local_files(pattern, recursive=False):
    return glob.glob(escape(pattern), recursive=recursive)
```

`hail_glob.py` is now a reusable implementation of the Hail Approved Glob Syntax. We can use it anywhere we need globbing like this:

```python3
import hail_glob

# ...

hail_glob.match(...)

# ...

hail_glob.local_files(...)

```